rtmr/rtmr.go

```diff
@@ -105,29 +106,23 @@ func (r *Extend) setRtmrIndex() error {
 // searchRtmrInterface searches for an rtmr entry in the configfs.
 func searchRtmrInterface(client configfsi.Client, index int) *Extend {
 	root := tsmRtmrPrefix
-	out := &Extend{}
-	if err := filepath.WalkDir(root, func(path string, d os.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-		if !d.IsDir() {
-			p, err := configfsi.ParseTsmPath(path)
-			if err != nil {
-				return err
-			}
+	var out *Extend
+	entries, err := client.ReadDir(root)
+	if err != nil {
+		return nil
+	}
+	for _, d := range entries {
+		if d.IsDir() {
 			r := &Extend{
 				RtmrIndex: index,
-				entry:     &configfsi.TsmPath{Subsystem: rtmrSubsystem, Entry: p.Entry},
+				entry:     &configfsi.TsmPath{Subsystem: rtmrSubsystem, Entry: d.Name()},
 				client:    client,
 			}
 			if r.validateIndex() {
 				out = r
-				return nil
+				break
 			}
 		}
-		return nil
-	}); err != nil || !out.validateIndex() {
-		return nil
 	}
 	return out
```

Comment:
This can become return nil and you can get rid of out.

Reply:
I see. Done